lnbits/wallets/breez_liquid.py

```diff
@@ -0,0 +1,289 @@
+            except Exception as exc:
+                logger.warning(exc)
+                raise ValueError(
+                    f"cannot initialize BreezLiquidSdkWallet: {exc!s}"
+                ) from exc
+
+        async def cleanup(self):
+            self.sdk_services.disconnect()
+
+        async def status(self) -> StatusResponse:
+            try:
+                info: breez_sdk.GetInfoResponse = self.sdk_services.get_info()
+            except Exception as exc:
+                return StatusResponse(f"Failed to connect to breez, got: '{exc}...'", 0)
+            return StatusResponse(None, int(info.wallet_info.balance_sat * 1000))
+
+        async def create_invoice(
+            self,
+            amount: int,
+            memo: Optional[str] = None,
+            description_hash: Optional[bytes] = None,
+            unhashed_description: Optional[bytes] = None,
+            **_,
+        ) -> InvoiceResponse:
+            try:
+                # TODO: issue with breez sdk, receive_amount is of type BITCOIN
+                # not ReceiveAmount after initialisation
+                receive_amount = breez_sdk.ReceiveAmount.BITCOIN(amount)
+                req = self.sdk_services.prepare_receive_payment(
+                    breez_sdk.PrepareReceiveRequest(
+                        payment_method=breez_sdk.PaymentMethod.BOLT11_INVOICE,
+                        amount=receive_amount,  # type: ignore
+                    )
+                )
+                receive_fees_sats = req.fees_sat
+
+                description = memo or (
+                    unhashed_description.decode() if unhashed_description else ""
+                )
+
+                res = self.sdk_services.receive_payment(
+                    breez_sdk.ReceivePaymentRequest(
+                        prepare_response=req,
+                        description=description,
+                        use_description_hash=description_hash is not None,
+                    )
+                )
+
+                bolt11 = res.destination
+                invoice_data = bolt11_decode(bolt11)
+                payment_hash = invoice_data.payment_hash
+
+                return InvoiceResponse(
+                    ok=True,
+                    checking_id=payment_hash,
+                    payment_request=bolt11,
+                    fee_msat=receive_fees_sats * 1000,
+                )
+            except Exception as e:
+                logger.warning(e)
+                return InvoiceResponse(ok=False, error_message=str(e))
+
+        async def pay_invoice(
+            self, bolt11: str, fee_limit_msat: int
+        ) -> PaymentResponse:
+            invoice_data = bolt11_decode(bolt11)
+
+            try:
+                prepare_req = breez_sdk.PrepareSendRequest(destination=bolt11)
+                req = self.sdk_services.prepare_send_payment(prepare_req)
+
+                fee_limit_sat = 50 + int(fee_limit_msat / 1000)
```

Comment:
what is the `50` magic number?
is this something that can be part of settings?

Reply:
yes i think it makes sense to make that configurable